src/Execute/ExecutionEnv.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2022 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Command} from '../Backend/Command';
+
+interface ExecutionEnv {
+  name(): string;
+  envInfo(): string;
+  runCommand(command: Command): Command;
+}
+
+// This is a base ExeuctionEnv.
+// on New ExecutionEnv implements, need to add 3 things
+// 1. ExecutionEnv class itself
+// 2. ExecutionEnv Creation function : named `function getEnv(name: string): ExecutionEnv`
```

Comment:
Can I ask a question?

Assume that we have, e.g, `RaspberryExecutionEnv` that inherits `ExecutionEnvBase`.
One way to create an instance is 

```
let r = new RaspberryExecutionEnv('raspberry 1');
```
and you're recommending

```
let r = getEnv('raspberry 1');
```

Could you explain why you designed like the latter?

Reply:
if we use like former one, then, every time some provider add new `ExecutionEnv`, they need to add some code that check and add this.(https://github.com/Samsung/ONE-vscode/pull/744#discussion_r891852735). so, I make a common factor as a module export function and use this.